Restoring support for parameters in media types for the sql query endpoint

cryptoe opened the pull request:
The PR: https://github.com/apache/druid/pull/17937 added new functionality to the sql query endpoints.




A bug was introduced with this PR where a sql query request going to the router with content type parameters such as ` Content-Type: application/json; text/plain` will always fail.

```
org.apache.druid.server.initialization.jetty.HttpException: Unsupported Content-Type: application/json; charset=UTF-8. Only application/json, text/plain or application/x-www-form-urlencoded is supported.
```

This patch restores the correct original behavior.

cryptoe commented:
@FrankChen021 I think we might need to do a 34.0.1 for this issue. 
Is this something you could drive ?

FrankChen021 reviewed (approved):
Thanks for the fix

cryptoe commented:
Adding more test cases. Would try to finish soon.

kfaraz commented:
> Let's discuss this on a dev list thread for 34.0.1, it's off topic for the current PR.

Sure, thanks!

gianm commented on sql/src/main/java/org/apache/druid/sql/http/SqlQuery.java:
```diff
@@ -343,5 +341,8 @@ private static SqlQuery from(
       throw new HttpException(Response.Status.BAD_REQUEST, "Unable to read query from request: " + e.getMessage());
     }
+    catch (IllegalArgumentException e) {
+      throw new HttpException(Response.Status.BAD_REQUEST, "Invalid Content-Type header: " + e.getMessage());
```
Not sure if this is right for this location. There's a lot going on in the `try` block and maybe something else can throw `IllegalArgumentException` other than an invalid content-type. Why not throw this exception directly in the `else` block of the content-type checking conditional?

cryptoe replied:
Thanks for the catch. Fixed it. The earlier catch location was an attempt to make switch statement work until I dropped that idea in the final cut .

gianm reviewed (commented):
LGTM other than a question about the exception handling. Thanks.

cryptoe commented:
Thanks for the review @FrankChen021 , @gianm . 
@gianm Addressed your concerns and merged it. Please feel free to comment if I missed anything. Will raise a follow up.
